public/app/features/commandPalette/useMatches.ts

```diff
@@ -201,35 +201,27 @@ function useInternalMatches(filtered: ActionImpl[], search: string): Match[] {
         const haystackItem = haystack[haystackIndex];
 
         // Use the position of the match as a stand-in for score
-        const substringPosition = haystackItem.toLowerCase().indexOf(query);
+        const substringPosition = haystackItem.indexOf(query);
 
         if (substringPosition > -1) {
-          const score = haystack.length - substringPosition;
+          const score = -substringPosition;
```

Comment:
i changed this to `substringPosition * -1` so hopefully its easier to catch, and added a comment explaining why we invert it for score